config.yaml

```diff
@@ -1,66 +1,76 @@
 logging:
-  debug: false
+  base:
+    debug: false
+  modules:
+    src.logserver.server:
+      debug: false
+    src.logcollector.batch_handler:
+      debug: false
+    src.logcollector.collector:
+      debug: false
+    src.prefilter.prefilter:
+      debug: false
+    src.inspector.inspector:
+      debug: false
+    src.detector.detector:
+      debug: false
 
-loglines:
-  fields:
-    - [ "timestamp", RegEx, '^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d{3}Z$' ]
-    - [ "status_code", ListItem, [ "NOERROR", "NXDOMAIN" ], [ "NXDOMAIN" ] ]
-    - [ "client_ip", IpAddress ]
-    - [ "dns_server_ip", IpAddress ]
-    - [ "domain_name", RegEx, '^(?=.{1,253}$)((?!-)[A-Za-z0-9-]{1,63}(?<!-)\.)+[A-Za-z]{2,63}$' ]
-    - [ "record_type", ListItem, [ "A", "AAAA" ] ]
-    - [ "response_ip", IpAddress ]
-    - [ "size", RegEx, '^\d+b$' ]
+pipeline:
+  stage_1: # Log Storage
```

Comment:
I would not name the stage 1,2,3,4.....
I prefer to use the module instead like you have implemented it for the logging

Reply:
So instead of `stage_1` we should use `log_storage`? We can also add a second layer `log_storage.logserver`.

Reply:
Or do you mean `src.logserver.server`?

Reply:
Both is fine for me, so either `log_storage` or `src.logserver.server`. Is a second layer necessary, like `log_storage.logserver`?
I think if we use the module path it is easier to add new modules/stages.

Reply:
For stages 4 and 5, I have the second layers `stage_4.inspector` and `stage_5.detector` to allow for additional modules in each stage, like for example in stage 2 with `collector` and `batch_sender`.

Having the layers for each stage is still good in my opinion, because each stage runs on its own Docker container (if Docker is used). Therefore, when a maintainer wants to change something in the LogCollection container, he can check the `stage_2` (or `LogCollection`) layer in the config.

Reply:
My suggestion is to rename `stage_1` and so on to the stages' names (i.e. `log_storage` etc.) and in each stage have the modules, either as a shortened name (`logserver`) or as the same names as for the logging (`src.logserver.server`).

Reply:
I see, that sounds good from my side.
To sum it up, we use the stages names (i.e. `log_storage`) as the top-layer and for each stage we use the module path (i.e. `src.logserver.server`). This allows us to add more modules (like `src.inspector.inpsector`, `src.inspector.inpsector1`, `src.inspector.inpsector2`) and we are still be able to differentiate and find.

Reply:
Yea, the top-layer would stay as `pipeline` but inside that section that is the plan

Reply:
I just committed the changes. I did shorten the names, however, because the long ones with the dots look a bit chunky, especially in the docs (e.g. `pipeline.log_storage.src.logserver.server`, which I now reduced to `pipeline.log_storage.logserver`).

Reply:
I wouldn't mind the naming scheme, shorten it is possible. However, I thought we can define different implementation types by that. In your case, we only refer to the package name. Is it still "chunky" without the `src`, like like`inspector.inpsector`, `inspector.inpsector1`, `inspector.inpsector2`?

Reply:
That would be possible as well. The question is if the proposed naming scheme is needed to distinguish the components. Your approach would look like that:
```
pipeline:
   ...
   data_inspection:
      inspector:
         inspector:
            ...
         inspector_1:
            ...
         inspector_2:
            ...
         ...
```
However, I think it is already easily distinguishable if we made it:
```
pipeline:
   ...
   data_inspection:
      inspector:
         ...
      inspector_1:
         ...
      inspector_2:
         ...
      ...
```
What's your opinion on that?

Reply:
Yes, indeed. I totally agree on that. I just wanted to make sure we cover every use-case for new add-ons.